caps/capabilities.go

```diff
@@ -118,12 +118,26 @@ func (r *Repository) Add(cap *Capability) error {
 	r.m.Lock()
 	defer r.m.Unlock()
 
+	// Reject capabilities with invalid names
 	if err := ValidateName(cap.Name); err != nil {
 		return err
 	}
+	// Reject capabilities with duplicate names
 	if _, ok := r.caps[cap.Name]; ok {
 		return fmt.Errorf("cannot add capability %q: name already exists", cap.Name)
 	}
+	// Reject capabilities with unknown types
+	for _, t := range r.types {
+		if cap.Type == t {
+			goto typeFound
```

Comment:
I guess this use of goto is sort of ok but it still does not quite feel right (unlike e.g. the C goto fail: pattern). 

You could do something like:
```
if !r.hasType(cap.Type) {...}

func (r *Repository) hasType(nt Type) bool {
	for _, t := range r.types {
		if nt == t {
			return true
		}
	}
	return false
}
```